regulation-worker/internal/delete/batch/batch.go

```diff
@@ -338,28 +338,28 @@ func (b *Batch) upload(ctx context.Context, uploadFileAbsPath, actualFileName, a
 	return nil
 }
 
-func (b *Batch) createPatternFile(userAttributes []model.UserAttribute) (string, error) {
+func (b *Batch) createPatternFile(userAttributes []model.User) (string, error) {
 	pkgLogger.Debug("creating a file with pattern to be searched & deleted.")
 
 	searchObject := make([]byte, 0)
 
 	for _, users := range userAttributes {
 		searchObject = append(searchObject, "/"...)
 		searchObject = append(searchObject, "\"userId\": *\""...)
-		searchObject = append(searchObject, users.UserID...)
+		searchObject = append(searchObject, users.ID...)
 		searchObject = append(searchObject, "\"/d;"...)
 
-		if users.Email != nil {
+		if users.Attributes["email"] != "" {
 			searchObject = append(searchObject, "/"...)
 			searchObject = append(searchObject, "\"email\": *\""...)
-			searchObject = append(searchObject, []byte(*users.Email)...)
+			searchObject = append(searchObject, []byte(users.Attributes["email"])...)
 			searchObject = append(searchObject, "\"/d;"...)
 		}
 
-		if users.Phone != nil {
+		if users.Attributes["phone"] != "" {
```

Comment:
Why don't we search by other attributes and use only email and phone?

Reply:
because for `batch` & `kv` destination type we only support deletion via `userId`, `Phone` & `email`. And, the reason behind this is, if the API caller gave a very common key & value that is present at rows which they didn't intended to delete can create problem. So, we are restricting to these three keys.